base/setup/reactos/drivepage.c

```diff
@@ -48,28 +48,261 @@ static const INT  column_alignment[MAX_LIST_COLUMNS] = {LVCFMT_LEFT, LVCFMT_LEFT
+        }
+
+        /* Copy over the valid ones */
+        *pch2 = *pch1;
+        ++pch2;
+    }
+    *pch2 = 0;
+
+    return bSanitized;
+}
+
+/**
+ * @brief
+ * Sanitize in-place any text found in the clipboard.
+ **/
+static BOOL
+DoSanitizeClipboard(
+    _In_ HWND hWnd)
+{
+    HGLOBAL hData;
+    LPWSTR pszText, pszSanitized;
+    BOOL bSanitized;
+
+    /* Protect read-only edit control from modification */
+    if (GetWindowLongPtrW(hWnd, GWL_STYLE) & ES_READONLY)
+        return FALSE;
+
+    if (!OpenClipboard(hWnd))
+        return FALSE;
+
+    hData = GetClipboardData(CF_UNICODETEXT);
+    pszText = GlobalLock(hData);
+    if (!pszText)
+    {
+        CloseClipboard();
+        return FALSE;
+    }
+
+    pszSanitized = _wcsdup(pszText);
+    GlobalUnlock(hData);
+    bSanitized = (pszSanitized && DoSanitizeText(pszSanitized));
+    if (bSanitized)
+    {
+        /* Update clipboard text */
+        DWORD cbData = (wcslen(pszSanitized) + 1) * sizeof(WCHAR);
+        hData = GlobalAlloc(GMEM_MOVEABLE | GMEM_SHARE, cbData);
+        pszText = GlobalLock(hData);
+        if (pszText)
+        {
+            CopyMemory(pszText, pszSanitized, cbData);
+            GlobalUnlock(hData);
+
+            SetClipboardData(CF_UNICODETEXT, hData);
+        }
+    }
+    free(pszSanitized);
+
+    CloseClipboard();
+    return bSanitized;
+}
+
+static VOID
+ShowErrorTip(
+    _In_ HWND hEdit)
+{
+    WCHAR szTitle[512];
+    WCHAR szText[512];
+
+    /* Load the resources */
+    LoadStringW(SetupData.hInstance, IDS_ERROR_INVALID_INSTALLDIR_CHAR_TITLE, szTitle, _countof(szTitle));
+    LoadStringW(SetupData.hInstance, IDS_ERROR_INVALID_INSTALLDIR_CHAR, szText, _countof(szText));
+
+    /* Show a warning balloon */
+    EDITBALLOONTIP balloon;
+    balloon.cbStruct = sizeof(balloon);
+    balloon.pszTitle = szTitle;
+    balloon.pszText  = szText;
+#if (_WIN32_WINNT < _WIN32_WINNT_VISTA)
+    balloon.ttiIcon  = TTI_ERROR;
+#else
+    balloon.ttiIcon  = 6; // TTI_ERROR_LARGE;
+#endif
+
+    MessageBeep(MB_ICONERROR);
+    Edit_ShowBalloonTip(hEdit, &balloon);
+
+    // NOTE: There is no need to hide it when other keys are pressed;
+    // the EDIT control will deal with that itself.
+}
+
+/**
+ * @brief
+ * Subclass edit window procedure to filter allowed characters
+ * for the ReactOS installation directory.
+ **/
+static LRESULT
+CALLBACK
+InstallDirEditProc(
+    _In_ HWND hWnd,
+    _In_ UINT uMsg,
+    _In_ WPARAM wParam,
+    _In_ LPARAM lParam)
+{
+    WNDPROC orgEditProc = (WNDPROC)GetWindowLongPtrW(hWnd, GWLP_USERDATA);
+
+    switch (uMsg)
+    {
+    case WM_UNICHAR:
+        if (wParam == UNICODE_NOCHAR)
+            return TRUE;
+        __fallthrough;
+
+    case WM_IME_CHAR:
+    case WM_CHAR:
+    {
+        WCHAR wch = (WCHAR)wParam;
+
+        /* Let the EDIT control deal with Control characters.
+         * It won't emit them as raw data in the text. */
+        if (wParam < ' ') // if (GetKeyState(VK_CONTROL) & 0x8000)
```

Comment:
```suggestion
        if (wParam < ' ')
```
or is that commented part a preparation for the future?

Reply:
This is for reviewers to tell me which one they think is the best to use.

Reply:
ok, then let's use whatever the reviewers will suggest, and strip the other one. If no suggestion would come in, then let's follow my suggestion to strip the comment, because what you actually have as active code today, is what you have tested so far already!